### PR TITLE
Fixing ODFE BWC links and upgrade to 2.2

### PR DESCRIPTION
**Pull request by sgup432**
**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Changing BWC link to point to OpenSearch 1.1 as opendistro is not supported officially.

Also upgrading to 2.2 version. - https://github.com/opensearch-project/performance-analyzer/issues/242

**Describe the solution you are proposing**

**Describe alternatives you've considered**

**Additional context**

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by sruti1312 — DISMISSED** [review]

**Review by peterzhuamazon — DISMISSED** [review]

**Review by sruti1312 — DISMISSED** [review]

File: src/main/java/org/opensearch/performanceanalyzer/http_action/config/RestConfig.java [hunk 1]
```diff
@@ -8,5 +8,5 @@
 /** Shared constants and configuration used by the REST controllers defined by this plugin. */
 public class RestConfig {
     public static final String PA_BASE_URI = "/_plugins/_performanceanalyzer";
-    public static final String LEGACY_PA_BASE_URI = "/_opendistro/_performanceanalyzer";
+    public static final String LEGACY_PA_BASE_URI = PA_BASE_URI;
```
Comment (sruti1312) [inline src/main/java/org/opensearch/performanceanalyzer/http_action/config/RestConfig.java:11]: We can remove this check. As legacy _opendistro API's are no more supported.

File: src/test/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java [hunk 2]
```diff
@@ -210,14 +210,9 @@ public void testGetAgentUri_WithHttps_WithActionsRedirection() throws Exception
...
-        String requestUri = "http://localhost:9200/_opendistro/_performanceanalyzer/_agent/invalid";
+        String requestUri = "http://localhost:9200/_plugins/_performanceanalyzer/_agent/invalid";
         RestRequest request = generateRestRequest(requestUri, "invalid");
         URL finalURI = performanceAnalyzerRp.getAgentUri(request);
         assertNull(finalURI);
-
-        requestUri = "http://localhost:9200/_plugins/_performanceanalyzer/_agent/invalid";
-        request = generateRestRequest(requestUri, "invalid");
-        finalURI = performanceAnalyzerRp.getAgentUri(request);
-        assertNull(finalURI);
```
Comment (sruti1312) [inline src/test/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java:221]: Why are we removing these?

**Comment by sruti1312** [comment]
Test failure as LEGACY endpoint is changed. We should maintain it as ```opendistro```

org.opensearch.performanceanalyzer.http_action.config.PerformanceAnalyzerClusterConfigActionTests > testLegacyUpdateLoggingSetting FAILED
    java.lang.IllegalArgumentException: Cannot replace existing handler for [/_plugins/_performanceanalyzer/cluster/config] for method: GET

File: src/main/java/org/opensearch/performanceanalyzer/http_action/config/RestConfig.java (hunk 1 above)
Comment (sgup432) [reply src/main/java/org/opensearch/performanceanalyzer/http_action/config/RestConfig.java]: Reverted back

File: src/test/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java (hunk 2 above)
Comment (sgup432) [reply src/test/java/org/opensearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java]: Reverted back

**Review by peterzhuamazon — APPROVED** [review]

**Review by bbarani — APPROVED** [review]

**Review by sruti1312 — APPROVED** [review]

**Pull request merged**